perf/benchmark/runner/runner.py

```diff
@@ -174,28 +169,50 @@ def run(self, conn, qps, size, duration):
         if self.extra_labels is not None:
             labels += "_" + self.extra_labels
 
-        grpc = ""
+        # TODO(oschaaf): For test, remove
+        duration = 1
+        
+        # Note: Labels is the last arg, and there's stuff depending on that.
+        # watch out when moving it.
+        nighthawk_args = [
+            "nighthawk_client", 
+            "--concurrency auto",
+            "--output-format json",
+            "--prefetch-connections",
+            "--open-loop",
+            "--experimental-h1-connection-reuse-strategy lru",
+            "--experimental-h2-use-multiple-connections",
+            "--nighthawk-service 127.0.0.1:{portmap}",
+            "--label Nighthawk",
+            "--connections {conn}",
+            "--rps {qps}",
+            "--duration {duration}",
+            "--request-header \"x-nighthawk-test-server-config: {{response_body_size:{size}}}\"",
+            "--label {labels}"
+        ]
+
+        # Our "gRPC" mode actually means:
+        #  - h2 
+        #  - with long running connections
+        #  - Also transfer request body sized according to "size".
         if self.mode == "grpc":
-            grpc = "-grpc -ping"
-
-        cacert_arg = ""
-        if self.cacert is not None:
-            cacert_arg = "-cacert {cacert_path}".format(cacert_path=self.cacert)
-
-        fortio_cmd = (
-            "fortio load -c {conn} -qps {qps} -t {duration}s -a -r {r} {cacert_arg} {grpc} -httpbufferkb=128 " +
-            "-labels {labels}").format(
-                conn=conn,
-                qps=qps,
-                duration=duration,
-                r=self.r,
-                grpc=grpc,
-                cacert_arg=cacert_arg,
-                labels=labels)
+            nighthawk_args.append("--h2")
+            if self.size:
+                nighthawk_args.append("--request-header \"content-length: {size}\"")
+
+        nighthawk_cmd = " ".join(nighthawk_args).format(
+            conn=conn,
+            qps=qps,
+            duration=duration,
+            labels=labels,
+            size=self.size,
+            portmap=NIGHTHAWK_GRPC_SERVICE_PORTMAP)
 
         if self.run_ingress:
```

Comment:
I think maybe for simplicity, this big if-clause can be extracted to a single function to call, since different modes share common command run structure. The only difference is the labels/modes. If you could, may be better to refactor these parts.

Reply:
Sure, done, refactored this for a bit. So far I've been trying to minimize changes for a bit to make review more fun :-)